internal/graph/check.go

```diff
@@ -974,22 +974,22 @@ func (c *LocalChecker) checkDirect(parentctx context.Context, req *ResolveCheckR
 }
 
 // checkComputedUserset evaluates the Check request with the rewritten relation (e.g. the computed userset relation).
-func (c *LocalChecker) checkComputedUserset(_ context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset_ComputedUserset) CheckHandlerFunc {
-	return func(ctx context.Context) (*ResolveCheckResponse, error) {
-		ctx, span := tracer.Start(ctx, "checkComputedUserset")
-		defer span.End()
+func (c *LocalChecker) checkComputedUserset(ctx context.Context, req *ResolveCheckRequest, rewrite *openfgav1.Userset_ComputedUserset) CheckHandlerFunc {
+	_, span := tracer.Start(ctx, "checkComputedUserset")
+	defer span.End()
 
-		if ctx.Err() != nil {
-			return nil, ctx.Err()
-		}
+	childRequest := clone(req)
 
-		rewrittenTupleKey := tuple.NewTupleKey(
-			req.GetTupleKey().GetObject(),
-			rewrite.ComputedUserset.GetRelation(),
-			req.GetTupleKey().GetUser(),
-		)
+	rewrittenTupleKey := tuple.NewTupleKey(
+		req.GetTupleKey().GetObject(),
+		rewrite.ComputedUserset.GetRelation(),
+		req.GetTupleKey().GetUser(),
+	)
 
-		return c.dispatch(ctx, req, rewrittenTupleKey)(ctx)
+	childRequest.TupleKey = rewrittenTupleKey
+
+	return func(ctx context.Context) (*ResolveCheckResponse, error) {
+		return c.ResolveCheck(ctx, childRequest)
```

Comment:
Why `c.ResolveCheck(ctx, childRequest)` and not `c.delegate.ResolveCheck(ctx, childRequest)`? If you run with `--check-cache-enabled` this means that we will not cache the result of a computed userset check. Is that what you wanted too?

Reply:
Here, this test fails in this PR, it never exits


```go
func TestDetectsCycle(t *testing.T) {
	ds := memory.New()
	t.Cleanup(ds.Close)

	storeID := ulid.Make().String()

	model := testutils.MustTransformDSLToProtoWithID(`
			model
				schema 1.1
			type user
			type document
				relations
					define x: x
					define y: x`)

	ctx := typesystem.ContextWithTypesystem(
		context.Background(),
		typesystem.New(model), // not NewAndValidate because this model isn't valid
	)

	ctx = storage.ContextWithRelationshipTupleReader(ctx, ds)

	checker := NewLocalChecker()
	t.Cleanup(checker.Close)

	resp, err := checker.ResolveCheck(ctx, &ResolveCheckRequest{
		StoreID:              storeID,
		AuthorizationModelID: model.GetId(),
		TupleKey:             tuple.NewTupleKey("document:1", "x", "user:maria"),
		RequestMetadata:      NewCheckRequestMetadata(20),
	})
	require.NoError(t, err)
	require.Nil(t, resp)
	require.False(t, resp.GetAllowed())
	require.True(t, resp.GetCycleDetected())
}
```

Reply:
Seems we are missing this test then.

Reply:
Added the test.